packages/myst-cli/src/process/site.ts

```diff
@@ -337,7 +337,11 @@ export function selectPageReferenceStates(
   const cache = castSession(session);
   let previousCounts: TargetCounts | undefined;
   const pageReferenceStates: ReferenceState[] = pages
-    .map(({ file }) => {
+    .map(({ file, hidden }: { file: string; hidden?: boolean }) => {
+      if (hidden) {
+        // Skip hidden files
+        return undefined;
```

Comment:
I think we just need to be less aggressive here... if `hidden`, skip the enumeration steps, but don't skip everything.